gabbi/httpclient.py

```diff
@@ -17,12 +17,32 @@
 import os
 import sys
 
-import httplib2
+import urllib3
 
 from gabbi import utils
 
 
-class VerboseHttp(httplib2.Http):
+class Http(urllib3.PoolManager):
+
+    def request(self, absolute_uri, method, body, headers, redirect):
+
+        retries = 1 if redirect else False
```

Comment:
The urllib3 Retry class lets you do all of that. It's remarkably configurable: [take a look](https://urllib3.readthedocs.org/en/latest/helpers.html#urllib3.util.retry.Retry).